pkg/host/host_linux.go

```diff
@@ -52,23 +55,35 @@ func isSupported(c *prog.Syscall, sandbox string) (bool, string) {
 	// Kallsyms seems to be the most reliable and fast. That's what we use first.
 	// If kallsyms is not present, we fallback to execution of syscalls.
 	kallsymsOnce.Do(func() {
-		kallsyms, _ = ioutil.ReadFile("/proc/kallsyms")
+		kallsyms, _ := ioutil.ReadFile("/proc/kallsyms")
+		if len(kallsyms) == 0 {
+			return
+		}
+		res := []string{
+			` T sys_([^\n]+)\n`, ` T ksys_([^\n]+)\n`,
+			` T __ia32_sys_([^\n]+)\n`, ` T __x64_sys_([^\n]+)\n`,
+			` T __arm64_sys_([^\n]+)\n`}
+		for _, re := range res {
+			rec := regexp.MustCompile(re)
+			matches := rec.FindAllStringSubmatch(string(kallsyms), -1)
```

Comment:
This will allocate and copy whole file for []byte->string conversion for each regexp. And later we will pin the whole string in the kallsymsSyscallSet map (I think regexp matches point into the input string).
I think it's better/faster to use `FindAllSubmatch` and then convert results to strings.